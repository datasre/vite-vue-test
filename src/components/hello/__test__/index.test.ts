/*
 * @Description  : 
 * @Version      : 1.0
 * @Author       : he_jian
 * @Date         : 2025-07-09 16:28:25
 * @LastEditors  : he_jian
 * @LastEditTime : 2025-07-09 16:36:34
 * @FilePath     : \\vite-vue-test\\src\\components\\hello\\__test__\\index.test.ts
 * Copyright (c) 2025 by he_jian, All Rights Reserved. 
 */
import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'

import Hello from '../index.vue'
describe('Hello', () => {
    it('挂载组件',async () => {
        expect(Hello).toBeTruthy()

        const wrapper = mount(Hello, {
            props: {
                count: 4
            }
        })

        await wrapper.get('button').trigger('click')

        expect(wrapper.text()).toContain('4 * 3 = 12')

        await wrapper.get('button').trigger('click')

        expect(wrapper.text()).toContain('4 * 4 = 16')
    })

    it('props', () => {
        const wrapper = mount(Hello, {
            props: {
                count: 6
            }
        })

        expect(wrapper.html()).toMatchSnapshot()
    })
})