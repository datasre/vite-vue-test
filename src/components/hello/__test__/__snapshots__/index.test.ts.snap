// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Hello > props 1`] = `
"<div>
  <div>6 * 2 = 12</div><button>Increment</button>
</div>"
`;
